### PR TITLE
Oracle scores during GFN training, pins and pairs sign, no_log_times and arg to reload GFN model each round

### PR DESCRIPTION
**Pull request by alexhernandezgarcia**
5 things here:
1. Functionality to log oracle scores during training of GFlowNet.
2. Fix issues with the positive sign of pairs and pins (this should be reviewed! Is it correct? Where else should we make changes?)
3. Argument to not log times to comet to avoid clutter
4. Argument to better control whether the GFN model from the previous round is loaded.
5. Argument and config to set the seed of the random number generator of GFlowNet

### CODE REVIEW

File: activeLearner.py [hunk 1]
```diff
@@ -537,7 +537,10 @@ def updateDataset(self, oracleSequences, oracleScores):
 
         if self.comet:
             self.comet.log_histogram_3d(dataset['scores'], name='dataset scores', step=self.pipeIter)
-            idx_sorted = np.argsort(dataset["scores"])
+            if any([s in self.config.dataset.oracle for s in ["pins", "pairs"]]):
+                idx_sorted = np.argsort(dataset["scores"])[::-1]
```
Comment (InfluenceFunctional) [inline activeLearner.py:541]: alternative to flipping this in the code is enforcing the nupack oracle always be defined on (-infty,0] - what do you think @alexhernandezgarcia

File: eval_al.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
+"""
+Double check that all initial data sets are equal.
```
Comment (InfluenceFunctional) [inline eval_al.py:2]: is this script just to print results of al runs?

File: eval_gflownet.py [hunk 3]
```diff
@@ -189,7 +189,10 @@ def main(args):
         print(f"\tMax score: {df_samples.scores.max()}")
         output_samples = workdir / "{}_samples_n{}.csv".format(model_alias, n_samples)
         df_samples.to_csv(output_samples)
-        scores_sorted = np.sort(df_samples["scores"].values)
+        if any([s in env.func for s in ["pins", "pairs"]]):
+            scores_sorted = np.sort(df_samples["scores"].values)[::-1]
```
Comment (InfluenceFunctional) [inline eval_gflownet.py:193]: same point on negative oracles

**Comment by InfluenceFunctional** [comment]
Looks good overall. If we want to flip the sign on pins, pairs, etc. I can do that throughout the code and make sure it runs clean.

File: activeLearner.py (hunk 1 above)
Comment (alexhernandezgarcia) [reply activeLearner.py:541]: Yes, that would be cleaner.

File: eval_al.py (hunk 2 above)
Comment (alexhernandezgarcia) [reply eval_al.py:2]: Oops, a few mistakes here:
1. This should have shown up on a different PR
2. The description is wrong.

This script is to compute some metrics from AL runs indeed. Very basic, can be extended.

Comment (InfluenceFunctional) [reply eval_al.py:2]: all good

**Pull request merged**